source/routing/tests/routing_api_test_data.h

```diff
@@ -496,6 +496,10 @@ struct routing_api_data {
         return speed_trait.to_duration(dist, mode);
     }
 
+    double to_precise_duration(float dist, navitia::type::Mode_e mode) {
```

Comment:
I'd add a `double navitia::time_duration::total_fractional_seconds() const` to `navitia::time_duration`. That way you can easily use that in tests and code.